crates/sui-framework/packages/move-stdlib/sources/macros.move

```diff
@@ -84,7 +84,7 @@ public macro fun num_to_string($x: _): String {
     buffer.to_string()
 }
 
-public macro fun range_do($start: _, $stop: _, $f: |_|) {
+public macro fun range_do<$R: drop>($start: _, $stop: _, $f: |_| -> $R) {
```

Comment:
I'm a bit torn, so my comment was more about starting a discussion than suggesting a particular direction. Mixing and matching _ and named generics feels a bit strange, but it doesn't really matter much for these since they are really just helpers. 
My point is `_` is neat, but it makes it harder to annotate types if you need to.

Reply:
Personally I am focused on the practical issue which is regular annoyance when I have a return value in a lambda. As to whether to use `_` or `$R: drop` - no opinion, but I find `_` slightly less confusing, and when someone looks at the macros as they are implemented right now, they might feel obliged to add (or at least to think) about `$R`, whereas the `_` is less misleading to less experienced devs.

Reply:
IMO in this case I think `_` should really be a type parameter due to the "fixity" of the types needed for `$start` `$stop` and `f: |_| -> $R` in particular all the `_`'s must be the same type. Whereas `_` for all of them provides a looser type signature that is not always correct.

Reply:
Fair enough. 
Let's swap all the `_` for `$T`. 

I think we should avoid mixing and matching `_` and `$T`, so if we should use `$T` here, let's do it